automation/workflow/workflow.go

```diff
@@ -0,0 +1,122 @@
+package workflow
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/qri-io/qri/profile"
+)
+
+var (
+	// ErrNotFound indicates that the workflow was not found in the store
+	ErrNotFound = fmt.Errorf("workflow not found")
+)
+
+// ID is a string identifier for a workflow
+type ID string
+
+// NewID creates a new workflow identifier
+func NewID() ID {
+	return ID(uuid.New().String())
+}
+
+// String returns the underlying id string
+func (id ID) String() string { return string(id) }
+
+// SetIDRand sets the random reader that NewID uses as a source of random bytes
+// passing in nil will default to crypto.Rand. This can be used to make ID
+// generation deterministic for tests. eg:
+//    myString := "SomeRandomStringThatIsLong-SoYouCanCallItAsMuchAsNeeded..."
+//    run.SetIDRand(strings.NewReader(myString))
+//    a := NewID()
+//    run.SetIDRand(strings.NewReader(myString))
+//    b := NewID()
+func SetIDRand(r io.Reader) {
+	uuid.SetRand(r)
+}
+
+// State holds all the information necessary to run a workflow
+type State struct {
+	ID        ID
+	DatasetID string
+	OwnerID   profile.ID
+	Deployed  bool
+}
+
+// Store manages & stores workflows, allowing listing and updating of workflows
+type Store interface {
+	Updater
+	// Lister
+}
+
+// An Updater can store and update workflows
+type Updater interface {
+	Create(did string, pid profile.ID) (*State, error)
+	Get(wid ID) (*State, error)
+	// Remove
+	// Update
+	// Deploy
+	// Undeploy
+}
+
+// A Lister lists entries from a workflow store
+type Lister interface {
+	// List
+	// ListDeployed
+}
+
+// NewDefaultStore returns the default expected store
+// TODO (ramfox): replace this with a file store asap, currently a MemStore
+func NewDefaultStore(ctx context.Context) (*MemStore, error) {
+	return NewMemStore(), nil
+}
```

Comment:
I don't think we need this function. I'd be better to just force users to pick the store type, and define a configuration-driven constructor once we have config in place